compose/runtime/runtime-lint/src/main/java/androidx/compose/runtime/lint/ComposableNamingDetector.kt

```diff
@@ -55,7 +56,7 @@ class ComposableNamingDetector : Detector(), SourceCodeScanner {
                 // special case where a generic return type and a Unit type parameter is used.
                 if (node.findSuperMethods().isNotEmpty()) return
 
-                val name = node.name
+                val name = (node.sourcePsi as? PsiNamedElement)?.name ?: node.name
```

Comment:
Can you please add a comment above the inlined implementation here to explain that it's synonymous to `nameFromSource`? We may consider switching back to the original API once we can upgrade Lint.

We also probably want to have that reference in case we need to take any changes from the upstream implementation.

Reply:
Would this be OK? Let me know if you have a better way of linking this.

```suggestion
                // NOTE: this is the inlined version of `UElement#nameFromSource` (available starting with Lint `31.10.0`)
                val name = (node.sourcePsi as? PsiNamedElement)?.name ?: node.name
```

Reply:
Yes, that looks good to me — anything that points to where this came from is fine.